src/patterns/strategy/examples.ts

```diff
@@ -0,0 +1,20 @@
+import {StrategyContext, FixedDiscount, PercentageDiscount} from './'
+
+const RANDOM_FIXED_DISCOUNT = 500;
+const RANDOM_PERCENTAGE_DISCOUNT = 15;
+const RANDOM_PRICE = 1000;
+
+const fixed = new FixedDiscount(RANDOM_FIXED_DISCOUNT);
+const percentage = new PercentageDiscount(RANDOM_PERCENTAGE_DISCOUNT);
+
+export const calculateFinalPrice = (price: number, strategy: FixedDiscount | PercentageDiscount): number => {
```

Comment:
In your example the `calculateFinalPrice` is a strategy client, so it should accept the strategy interface instead of strategies implementations.
```suggestion
export const calculateFinalPrice = (price: number, strategy: IBaseDiscountStrategy): number => {
```